scalactic-test/src/test/scala/org/scalactic/anyvals/RegexStringSpec.scala

```diff
@@ -0,0 +1,584 @@
+        an [AssertionError] should be thrownBy RegexString.ensuringValid("99)")
+      }
+    }
+    describe("should offer an isValid predicate method that") {
+      it("returns true if the passed String is a valid regex") {
+        RegexString.isValid("") shouldBe true
+        RegexString.isValid(".") shouldBe true
+        RegexString.isValid("a+") shouldBe true
+        RegexString.isValid("[abc]") shouldBe true
+        RegexString.isValid("(a|b)") shouldBe true
+        RegexString.isValid("false") shouldBe true
+      }
+      it("returns false if the passed String is not a valid regex") {
+        RegexString.isValid("a|b)") shouldBe false
+        RegexString.isValid("*") shouldBe false
+        RegexString.isValid("+") shouldBe false
+        RegexString.isValid("?") shouldBe false
+        RegexString.isValid("*true") shouldBe false
+      }
+    } 
+    describe("should offer a fromOrElse factory method that") {
+      it("returns a RegexString if the passed regex is valid") {
+        RegexString.fromOrElse("50", RegexString("42")).value shouldBe "50"
+        RegexString.fromOrElse("100", RegexString("42")).value shouldBe "100"
+      }
+      it("returns a given default if the passed regex is not valid") {
+        RegexString.fromOrElse(")", RegexString("42")).value shouldBe "42"
+        RegexString.fromOrElse("+", RegexString("42")).value shouldBe "42"
+        RegexString.fromOrElse("(a|b", RegexString("42")).value shouldBe "42"
+      }
+    } 
+    it("should offer an ensuringValid method that takes a String => String, throwing AssertionError if the result is invalid") {
+      RegexString("33").ensuringValid(s => (s.toInt + 1).toString) shouldEqual RegexString("34")
+      an [AssertionError] should be thrownBy { RegexString("33").ensuringValid(_ + "(") }
+    }
+    it("should offer a length method that is consistent with String") {
+      forAll { (regStr: RegexString) =>
+        regStr.length shouldEqual regStr.value.length
+      }
+    }
+    it("should offer a charAt method that is consistent with String") {
+      forAll { (regStr: RegexString, pint: PosInt) =>
+        whenever (regStr.length > 0) {
+          val idx = pint % regStr.length
+          regStr.charAt(idx) shouldEqual regStr.value.charAt(idx)
+        }
+      }
+    }
+    it("should offer a codePointAt method that is consistent with String") {
+      forAll { (regStr: RegexString, pint: PosInt) =>
+        whenever (regStr.length > 0) {
+          val idx = pint % regStr.length
+          regStr.codePointAt(idx) shouldEqual regStr.value.codePointAt(idx)
+        }
+      }
+    }
+    it("should offer a codePointBefore method that is consistent with String") {
+      forAll { (regStr: RegexString, pint: PosInt) =>
+        whenever (regStr.length > 0) {
+          val idx = (pint % regStr.length) + 1
+
+          regStr.codePointBefore(idx) shouldEqual
+            regStr.value.codePointBefore(idx)
```

Comment:
i think it can't work as you want

line 132 - `shouldEqual` returns `Assertion`
line 133 - result of line 132 is ignored, and this line returns `Int`